src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3295,21 +3295,10 @@ private TextSpan GetThisParameterDiagnosticSpan(ISymbol member)
 
         private TextSpan GetVariableDiagnosticSpan(ISymbol local)
         {
-            return local.Locations.First().SourceSpan;
-        }
-
-        private static ImmutableArray<IParameterSymbol> GetParametersWithSyntax(ISymbol member)
-        {
-            var method = (IMethodSymbol)member;
-
-            if (method.AssociatedSymbol != null)
-            {
-                return ((IPropertySymbol)method.AssociatedSymbol).Parameters;
-            }
-            else
-            {
-                return method.Parameters;
-            }
+            // Note that in VB implicit value parameter in property setter doesn't have a location.
+            // In C# its location is the location of the setter.
+            // See https://github.com/dotnet/roslyn/issues/14273
+            return local.Locations.FirstOrDefault()?.SourceSpan ?? local.ContainingSymbol.Locations.First().SourceSpan;
```

Comment:
This is in AbstractEditAndContinueAnalyzer.  So why not have two subclasses of this (for VB and C#) and  have hte language specific cases handled there?